ethereum/eth/src/test/java/tech/pegasys/pantheon/ethereum/eth/sync/fullsync/IncrementerTest.java

```diff
@@ -0,0 +1,111 @@
+import tech.pegasys.pantheon.metrics.prometheus.PrometheusMetricsSystem;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+
+public class IncrementerTest {
+  private ProtocolSchedule<Void> protocolSchedule;
+  private EthProtocolManager ethProtocolManager;
+  private EthContext ethContext;
+  private ProtocolContext<Void> protocolContext;
+  private SyncState syncState;
+
+  private BlockchainSetupUtil<Void> localBlockchainSetup;
+  private MutableBlockchain localBlockchain;
+  private BlockchainSetupUtil<Void> otherBlockchainSetup;
+  private Blockchain otherBlockchain;
+  private MetricsSystem metricsSystem = new PrometheusMetricsSystem();
+
+  @Test
+  public void parallelDownloadPipelineCounterShouldIncrement() {
+    localBlockchainSetup = BlockchainSetupUtil.forTesting();
+    localBlockchain = spy(localBlockchainSetup.getBlockchain());
+    otherBlockchainSetup = BlockchainSetupUtil.forTesting();
+    otherBlockchain = otherBlockchainSetup.getBlockchain();
+
+    protocolSchedule = localBlockchainSetup.getProtocolSchedule();
+    protocolContext = localBlockchainSetup.getProtocolContext();
+    ethProtocolManager =
+        EthProtocolManagerTestUtil.create(
+            localBlockchain,
+            localBlockchainSetup.getWorldArchive(),
+            new EthScheduler(1, 1, 1, new NoOpMetricsSystem()));
+    ethContext = ethProtocolManager.ethContext();
+    syncState = new SyncState(protocolContext.getBlockchain(), ethContext.getEthPeers());
+
+    otherBlockchainSetup.importFirstBlocks(15);
+    final long targetBlock = otherBlockchain.getChainHeadBlockNumber();
+    // Sanity check
+    assertThat(targetBlock).isGreaterThan(localBlockchain.getChainHeadBlockNumber());
+
+    final RespondingEthPeer peer =
+        EthProtocolManagerTestUtil.createPeer(ethProtocolManager, otherBlockchain);
+    final RespondingEthPeer.Responder responder =
+        RespondingEthPeer.blockchainResponder(otherBlockchain);
+
+    final SynchronizerConfiguration syncConfig =
+        SynchronizerConfiguration.builder().downloaderChainSegmentSize(10).build();
+    final FullSyncDownloader<?> downloader = downloader(syncConfig);
+    downloader.start();
+
+    peer.respondWhileOtherThreadsWork(responder, () -> !syncState.syncTarget().isPresent());
+    assertThat(syncState.syncTarget()).isPresent();
+    assertThat(syncState.syncTarget().get().peer()).isEqualTo(peer.getEthPeer());
+
+    peer.respondWhileOtherThreadsWork(
+        responder, () -> localBlockchain.getChainHeadBlockNumber() < targetBlock);
+
+    assertThat(localBlockchain.getChainHeadBlockNumber()).isEqualTo(targetBlock);
+
+    final List<Observation> metrics =
+        metricsSystem.getMetrics(MetricCategory.SYNCHRONIZER).collect(Collectors.toList());
+    for (Observation observation : metrics) {
+      if (observation.getMetricName().equals("outboundQueueCounter")) {
+        assertThat(observation.getValue()).isEqualTo(5.0);
```

Comment:
Hmm.... why aren't the inbound and outbound counts equal?  Why is there one more inbound?